### PR TITLE
Median tract profile

### PR DESCRIPTION
**Pull request by 36000**
gives user option to calculate the median tract profile, as well as provide their own weighting scheme.

### CODE REVIEW

File: AFQ/api.py [hunk 1]
```diff
@@ -407,6 +417,18 @@ def __init__(self,
             raise TypeError(
                 "mapping must be a mapping defined"
                 + " in `AFQ.definitions.mapping`")
+        if not (profile_weights is None
+                or isinstance(profile_weights, str)
+                or callable(profile_weights)
+                or hasattr(profile_weights, "__len__")):
+            raise TypeError(
+                "profile_weights must be string, None, callable, or"
+                + "a 1D or 2D array")
```
Comment (arokem) [inline AFQ/api.py:426]: That's very restrictive! 😄

File: AFQ/api.py [hunk 2]
```diff
@@ -1353,11 +1378,32 @@ def _tract_profiles(self, row):
...
+                                values = np.array(
+                                    values_from_volume(
+                                        scalar_data,
+                                        fgarray,
+                                        row["dwi_affine"]))
+                                weights = np.zeros(values.shape)
+                                for ii, jj in enumerate(
+                                    np.argsort(values, axis=0)[
+                                        len(values) // 2, :]):
+                                    weights[jj, ii] = 1
```
Comment (arokem) [inline AFQ/api.py:1397]: This also seems like a potentially useful way to find this: https://stackoverflow.com/a/64015677/3532933

**Review by arokem — COMMENTED** [review]
Thanks for the quick implementation! Does it work as expected? I realize that this is a bit hard to test. Maybe worth pushing this upstream into dipy's `afq_profile`?

**Comment by 36000** [comment]
It works as expected. I think in DIPY I will wait for next week to do the PR, and there I will just calculate the median directly.

**Pull request merged**